### DISCUSSION TITLE
Version 0.2?

### DISCUSSION THREAD

**Discussion by macrumpel** [jivanpal/apfs-tools#45] [General]
Hello @jivanpal 
The documentation is for 0.2 but I only find .13. 
i am stuck with my Fusion Drive and all the data. Inspection blocks at ephemeral object. Any chance to have the new version soon?
Disk drill find my files but recovery gives completely corrupted documents 
Thank you so much

Update : I get this error: on list /dev/disk0s2 0 /User/Documents

```
Opening file at `/dev/disk0s2` in read-only mode ... OK.
Simulating a mount of the APFS container.
Validating checksum of block 0x0 ... OK.
Locating the checkpoint descriptor area:
- Its length is 304 blocks.
- It is contiguous.
- The address of its first block is 0x32bae.
Loading the checkpoint descriptor area into memory ... OK.
Locating the most recent well-formed container superblock in the checkpoint descriptor area:
- It lies at index 117 within the checkpoint descriptor area.
- The corresponding checkpoint starts at index 116 within the checkpoint descriptor area, and spans 2 blocks.

Loading the corresponding checkpoint ... OK.
- There are 11 checkpoint-mappings in this checkpoint.
Reading the Ephemeral objects used by this checkpoint ... OK.
Validating the Ephemeral objects ... FAILED.
An Ephemeral object used by this checkpoint is malformed. Going back to look at the previous checkpoint instead.
END: Handling of this case has not yet been implemented.
```

**Comment by jivanpal** [comment]
Hi, the documentation is for the version that is currently in development (albeit very, very slowly over the past year).  Version 0.1.3 and earlier have different command syntax that is documented in the program's help text. Based on the rate at which I'm currently making progress on v0.2.0, I'd say it'll be out around June 2023. If I can find the time again to work on it more consistently, I can see it coming out a few months sooner.

I haven't got support for Fusions Drives on the roadmap, but it's something I'll add. Crypto support will likely be added first; I don't believe many people use APFS with a Fusion Drive setup (it is really designed for use exclusively with SSDs), and I don't think much headway has been made by other people on reverse-engineering it.

> Disk drill find my files but recovery gives completely corrupted documents

If you've done a deep scan of the drivers, this sounds like you're probably out of luck completely then. However, I don't know what Disk Drill's capabilities are when it comes to Fusion Drives, so it may just be lacking in that area.

> ```
> Validating the Ephemeral objects ... FAILED.
> An Ephemeral object used by this checkpoint is malformed. Going back to look at the previous checkpoint instead.
> END: Handling of this case has not yet been implemented.
> ```

I really need to get rid of this Ephemeral object validation / exit on failure behaviour, it's not really useful. Someone else asked about this, so I created the branch `ignore-ephemerals` for the time being to skip this check. Do `git checkout ignore-ephemerals`, then `make`, to compile that version, then you can try `drat inspect` or `drat list` again.
